lib/Doctrine/DBAL/Connection.php

```diff
@@ -501,58 +443,58 @@ public function setFetchMode($fetchMode) : void
      * Prepares and executes an SQL query and returns the first row of the result
      * as an associative array.
      *
-     * @param string         $statement The SQL query.
-     * @param mixed[]        $params    The query parameters.
-     * @param int[]|string[] $types     The query parameter types.
+     * @param string                                           $query  The SQL query.
+     * @param array<int, mixed>|array<string, mixed>           $params The prepared statement params.
+     * @param array<int, int|string>|array<string, int|string> $types  The query parameter types.
```

Comment:
Could this be written as `array<int|string, int|string>`?

Reply:
Not really. It's either an array with integer keys corresponding to parameter positions or an array with string keys corresponding to their names. Not a mix.